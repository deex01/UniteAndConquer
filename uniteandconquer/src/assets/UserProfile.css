td,
th {
  border: 10px solid #ffd9a0;
  box-sizing: border-box;
}

table {
  border-collapse: collapse;
}

th {
  font-size: 25px;
}

td {
  max-width: 700px;
  padding: 15px;
}

.post-title {
  font-size: 24px;
  margin-bottom: 20px;
}

.post-content {
  font-size: 17px;
}

.user-profile-page {
  display: grid;
  grid-template-columns: 0.3fr 0.6fr;
}

.profile-title {
  font-size: 25px;
  text-align: center;
}

.notifications-pos {
  position: absolute;
  top: 0;
  right: 0;
}

.bell-pos {
  position: absolute;
  top: 0;
  right: 0;
  padding: 50px;
}

.bell-button {
  background: none;
  border: none;
}

.bell-button:hover {
  color: rgb(133, 235, 85);
}