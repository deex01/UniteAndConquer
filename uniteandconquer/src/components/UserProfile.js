import React, { useState } from 'react';
import SidebarTags from './SidebarTags';
import Notifications from './Notifications';
import '../assets/UserProfile.css';

function UserProfile() {
  const [showNotifs, setShowNotifs] = useState(false);
  return (
    <div className="user-profile-page">
      <div>{SidebarTags()}</div>
      <div>
        <div className="profile-title"><h1>My Profile</h1></div>
        <div className="bell-pos">
          <button data-testid="bell-button" className="bell-button" type="button" onClick={() => setShowNotifs(!showNotifs)}>
            {' '}
            <i className="fas fa-bell fa-2x" />
          </button>
        </div>
        <div className="notifications-pos">{showNotifs ? Notifications({ showNotifs, setShowNotifs }) : ''}</div>
        <div>
          <table>
            <tbody>
              <tr>
                <th>My Active Posts</th>
                <th>My Wishlist</th>
              </tr>
              <tr>
                <td>
                  <div className="post-title">Post [ID]: [itemName] - [status]</div>
                  <div className="post-content">
                    This post is led by [ownerName] and trades [itemNumTarget] of [itemCurrent] for
                    $[pricePerItem]
                    <br />
                    <br />
                    [tags]
                  </div>
                </td>
                <td>
                  <div className="post-title">Post [ID]: [itemName] - [status]</div>
                  <div className="post-content">
                    This post is led by [ownerName] and trades [itemNumTarget] of [itemCurrent] for
                    $[pricePerItem]
                    <br />
                    <br />
                    [tags]
                  </div>
                </td>
              </tr>
              <tr>
                <td>
                  <div className="post-title">Post [ID]: [itemName] - [status]</div>
                  <div className="post-content">
                    This post is led by [ownerName] and trades [itemNumTarget] of [itemCurrent] for
                    $[pricePerItem]
                    <br />
                    <br />
                    [tags]
                  </div>
                </td>
                <td>
                  <div className="post-title">Post [ID]: [itemName] - [status]</div>
                  <div className="post-content">
                    This post is led by [ownerName] and trades [itemNumTarget] of [itemCurrent] for
                    $[pricePerItem]
                    <br />
                    <br />
                    [tags]
                  </div>
                </td>
              </tr>
              <tr>
                <td>
                  <div className="post-title">Post [ID]: [itemName] - [status]</div>
                  <div className="post-content">
                    This post is led by [ownerName] and trades [itemNumTarget] of [itemCurrent] for
                    $[pricePerItem]
                    <br />
                    <br />
                    [tags]
                  </div>
                </td>
                <td>
                  <div className="post-title">Post [ID]: [itemName] - [status]</div>
                  <div className="post-content">
                    This post is led by [ownerName] and trades [itemNumTarget] of [itemCurrent] for
                    $[pricePerItem]
                    <br />
                    <br />
                    [tags]
                  </div>
                </td>
              </tr>
              <tr>
                <td>
                  <div className="post-title">Post [ID]: [itemName] - [status]</div>
                  <div className="post-content">
                    This post is led by [ownerName] and trades [itemNumTarget] of [itemCurrent] for
                    $[pricePerItem]
                    <br />
                    <br />
                    [tags]
                  </div>
                </td>
                <td>
                  <div className="post-title">Post [ID]: [itemName] - [status]</div>
                  <div className="post-content">
                    This post is led by [ownerName] and trades [itemNumTarget] of [itemCurrent] for
                    $[pricePerItem]
                    <br />
                    <br />
                    [tags]
                  </div>
                </td>
              </tr>

            </tbody>
          </table>

        </div>

      </div>
    </div>
  );
}

export default UserProfile;
