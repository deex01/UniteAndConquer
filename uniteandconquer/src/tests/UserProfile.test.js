/**
 * @jest-environment jsdom
 */

import React from 'react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import {
  render, screen, fireEvent,
} from '@testing-library/react';
import UserProfile from '../components/UserProfile';

const mockedUsedLink = jest.fn();
const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  Link: () => mockedUsedLink,
  useNavigate: () => mockedUsedNavigate,
}));

test('profile title header', () => {
  render(<UserProfile />);
  const linkElement = screen.getByText('My Profile');
  expect(linkElement).toBeInTheDocument();
});

test('push to show notifications', () => {
  render(<UserProfile />);
  fireEvent.click(screen.getByTestId('bell-button'));
  const linkElement = screen.getByText('Notifications');
  expect(linkElement).toBeInTheDocument();
  fireEvent.click(screen.getByTestId('notif-button'));
});

test('snapshot user-profile', () => {
  const component = renderer.create(<UserProfile />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
