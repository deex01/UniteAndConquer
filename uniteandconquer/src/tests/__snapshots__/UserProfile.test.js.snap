// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot user-profile 1`] = `
<div
  className="user-profile-page"
>
  <div>
    <div
      className="sidebar"
    >
      <div
        className="greeting"
      >
        <i
          className="far fa-user-circle fa-2x"
        />
         
        Hi, Jeremy
      </div>
      <div
        className="home"
      />
      <div
        className="settings"
      />
      <div
        className="preferences"
      >
        <div
          className="preferences-title"
        >
          My Preferences
        </div>
        <div
          className="preference-tag"
        >
          Tag1
        </div>
        <div
          className="preference-tag"
        >
          Tag2
        </div>
        <div
          className="preference-tag"
        >
          Tag3
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="profile-title"
    >
      <h1>
        My Profile
      </h1>
    </div>
    <div
      className="bell-pos"
    >
      <button
        className="bell-button"
        data-testid="bell-button"
        onClick={[Function]}
        type="button"
      >
         
        <i
          className="fas fa-bell fa-2x"
        />
      </button>
    </div>
    <div
      className="notifications-pos"
    >
      
    </div>
    <div>
      <table>
        <tbody>
          <tr>
            <th>
              My Active Posts
            </th>
            <th>
              My Wishlist
            </th>
          </tr>
          <tr>
            <td>
              <div
                className="post-title"
              >
                Post [ID]: [itemName] - [status]
              </div>
              <div
                className="post-content"
              >
                This post is led by [ownerName] and trades [itemNumTarget] of [itemCurrent] for $[pricePerItem]
                <br />
                <br />
                [tags]
              </div>
            </td>
            <td>
              <div
                className="post-title"
              >
                Post [ID]: [itemName] - [status]
              </div>
              <div
                className="post-content"
              >
                This post is led by [ownerName] and trades [itemNumTarget] of [itemCurrent] for $[pricePerItem]
                <br />
                <br />
                [tags]
              </div>
            </td>
          </tr>
          <tr>
            <td>
              <div
                className="post-title"
              >
                Post [ID]: [itemName] - [status]
              </div>
              <div
                className="post-content"
              >
                This post is led by [ownerName] and trades [itemNumTarget] of [itemCurrent] for $[pricePerItem]
                <br />
                <br />
                [tags]
              </div>
            </td>
            <td>
              <div
                className="post-title"
              >
                Post [ID]: [itemName] - [status]
              </div>
              <div
                className="post-content"
              >
                This post is led by [ownerName] and trades [itemNumTarget] of [itemCurrent] for $[pricePerItem]
                <br />
                <br />
                [tags]
              </div>
            </td>
          </tr>
          <tr>
            <td>
              <div
                className="post-title"
              >
                Post [ID]: [itemName] - [status]
              </div>
              <div
                className="post-content"
              >
                This post is led by [ownerName] and trades [itemNumTarget] of [itemCurrent] for $[pricePerItem]
                <br />
                <br />
                [tags]
              </div>
            </td>
            <td>
              <div
                className="post-title"
              >
                Post [ID]: [itemName] - [status]
              </div>
              <div
                className="post-content"
              >
                This post is led by [ownerName] and trades [itemNumTarget] of [itemCurrent] for $[pricePerItem]
                <br />
                <br />
                [tags]
              </div>
            </td>
          </tr>
          <tr>
            <td>
              <div
                className="post-title"
              >
                Post [ID]: [itemName] - [status]
              </div>
              <div
                className="post-content"
              >
                This post is led by [ownerName] and trades [itemNumTarget] of [itemCurrent] for $[pricePerItem]
                <br />
                <br />
                [tags]
              </div>
            </td>
            <td>
              <div
                className="post-title"
              >
                Post [ID]: [itemName] - [status]
              </div>
              <div
                className="post-content"
              >
                This post is led by [ownerName] and trades [itemNumTarget] of [itemCurrent] for $[pricePerItem]
                <br />
                <br />
                [tags]
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
